tensorflow/core/kernels/resize_bilinear_op_gpu.cu.cc

```diff
@@ -25,12 +25,78 @@ limitations under the License.
 #include "tensorflow/core/platform/types.h"
 #include "tensorflow/core/util/gpu_kernel_helper.h"
 
+// auxilary 16-byte datatype for ResizeBilinearKernel_faster
+// the fields are not important. The only purpose of this is to read 16 bytes
+// from GPU gloal memory
+struct four_floats{
```

Comment:
Could you use float4 instead?

Reply:
you are referring to the built-in type `float4` right?

An interesting thing: when I replaced `four_floats` with `float4`, the speed about doubled.
For comparison, with batch = 1, channel = 64, input height = 128, input width = 128, height scale 1/16, width scale = 1/16, the speed is ( on my GTX 1080 ):
ref = 13188ms
four_floats (the older version which I PR-ed) = 9149ms
float4 (replaced `four_floats` with `float4` ) = 5166ms 

so compared to the original, the speed nearly tripled :) HUGE Thanks to your suggestion!!

Reply:
I think by default this structure is only 4 bytes aligned. But I'm not sure if aligning it is sufficient. The float4 also uses 'builtin': `struct __device_builtin__ __builtin_align__(16) float4`